Automattic/jetpack
# CRM: Create Automations UI root

## Proposed changes:


Fixes https://github.com/Automattic/zero-bs-crm/issues/3179

This PR adds the root of a React app for Automations and creates a new Automations module to load and serve it.

<img width="1171" alt="image" src="https://github.com/Automattic/jetpack/assets/42627630/7cd440a3-d7aa-40c0-a51a-782cbd28f8d9">

**It aim to do so by:**

* Register "Automations" as a core module
* Add feature flag to prevent the UI to be exposed.
* Finish the CRM specific `AdminPage` component
  * We also correct an older dummy test for the OBW component that was used to test if React builds worked and was tested.
* Add a new `<AutomationAdmin>` component that will be the main component for the automations view.

**Caveats:**

* The RNA container component only supports the narrow Jetpack view - not the ~1551px one that was recently used in Stats and CRM. I will create a separate PR to address this issue in the RNA package.

### Other information:


## Jetpack product discussion


p1HpG7-mCz-p2

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






1. Add the following to a mu-plugin (e.g. `0-sandbox.php`): `add_filter( 'jetpack_crm_feature_flag_automations', '__return_true' );`
2. Run `pnpm build` in the CRM folder.
3. Visit `/wp-admin/admin.php?page=jpcrm-automations` and verify that you see a page with an "Automations" header like the screenshot above.
4. In the CRM folder run `pnpm run test` and verify that all Automations tests pass.
5. Verify that Automations shows up in the menu bar.
6. Disable the feature flag filter and verify that Automations does not show up in the menu bar and that the page is not accessible.
